tools/dump_tool/dump_tool.c

```diff
@@ -389,28 +453,86 @@ int main(int argc, char *argv[])
 		goto tty_lock_err;
 	}
 
-	if (do_handshake(dev_fd) != 0) {
+	printf("Target device locked and ready to DUMP!!!\n");
+
+	printf("Chose from the following options:-\n1. RAM Dump\n2. Userfs Dump\n3. Both Dumps\n4. Exit Dump Tool\n");
+	fflush(stdout);
+	scanf("%d", &choice);
+	getchar();
+
+	if (choice == 2) {
+		printf("Skipping RAM Dump\n");
```

Comment:
How to skip FS dump? Even user select `1. RAM DUMP`, the device goes to fs dump state after ram dump. Could you check that?
And How about making infinite loop in device to get the user choice contiuousely?
It supports after getting RAM dump, reconnect and get another dump or the same dump again.
```
while (1) {
  	/* Receive hanshake string from HOST */
	do {
		host_buf[0] = up_getc();
	} while (host_buf[0] != target_str[0]);

	for (i = 1; i < HANDSHAKE_STR_LEN_MAX; ) {
		if ((ch = up_getc()) != -1) {
			host_buf[i] = ch;
			i++;
		}
	}

	/* Check handshake string against RAMDUMP HANDSHAKE string */
	if (strncmp(host_buf, RAMDUMP_HANDSHAKE_STRING, strlen(RAMDUMP_HANDSHAKE_STRING) + 1) == 0) {
		up_lowputc('A');
	} else {
		if (strncmp(host_buf, FSDUMP_HANDSHAKE_STRING, strlen(FSDUMP_HANDSHAKE_STRING) + 1) == 0) {
		/* Check handshake string against FSDUMP HANDSHAKE string */
			goto filesystem_dump;
		} else {
			/* Send NAK */
			up_lowputc('N');
		}
	}

	do RAMdump or FSdump...
}
```

Reply:
If user enters '1' for Ramdump, the tool skips FS Dump after Ramdump and goes straight to to TOOL EXIT:
Line 508 causes tool to exit after skipping FS Dump.

Reply:
Yeah, tool skips the fsdump with user input '1'. But device waits the fsdump message. That's why I said `the device goes to fs dump state after ram dump.`

Reply:
I added the character 'F' to serve as a handshake initiated from the TARGET side. It means the TARGET is now available to dump Userfs.
All other handshakes are initiated from HOST side.
So this character helps in synchronization after RAMDUMP has been sent.
That is why in case only FS DUMP is needed, then this synchronization character is skipped.

Reply:
Initial attempt was to have a loop as you suggested, but we observed the following: -
1. Synchronization of send and receive was even more tricky with an infinite choice loop.
2. Since there are only 2 dumps (RAM and FS), maximum loop iterations will be 2 only. So a linear interface seemed simpler.
3. User has to wait for 1st dump to complete to request the next dump. As you suggested earlier, the list of options solves this problem.